pcs/lib/tools.py

```diff
@@ -76,13 +78,15 @@ def dict_to_environment_file(config_dict):
     return "".join(lines)
 
 
-def write_tmpfile(data, binary=False):
+def write_tmpfile(
+    data: Union[None, bytes, str], binary: bool = False
+) -> Union[IO[str], IO[bytes]]:
```

Comment:
By using [@typing.overload](https://docs.python.org/3/library/typing.html#typing.overload) we can avoid casting its result like in this [case](https://github.com/ClusterLabs/pcs/pull/602/files#diff-e0413fa653ac3d6f221059dbeaa40db262b65694eb74a3f5133cddc574ce467fR129)